pyfr/plugins/tavg.py

```diff
@@ -351,3 +357,198 @@ def __call__(self, intg):
                     self.tstart_acc = intg.tcurr
 
                 self.tout_last = intg.tcurr
+
+
+class TavgCLIPlugin(BaseCLIPlugin):
+    name = 'tavg'
+    systems = ['*']
+    formulations = ['dual', 'std']
+
+    @classmethod
+    def add_cli(cls, parser):
+        cmd = cls.name
+        cp = parser.add_parser(cmd, help=f'{cmd} --help')
+        sp = cp.add_subparsers(dest='sub_cmd')
+
+        # Merge command
+        scmd = 'merge'
+        ssp = sp.add_parser(scmd, help=f'{cmd} {scmd} --help')
+        ssp.set_defaults(process=TavgCLIPlugin.merge_cli)
+        ssp.add_argument('solns', nargs='*', help='Averages to merge')
+        ssp.add_argument('output', help='Output file name')
+
+    @cli_external
+    def merge_cli(self):
+        # Open all the solution files
+        self._prepare_files(self.args.solns)
+
+        # Initialise things needed for the merge
+        self._init_tavg_merge()
+
+        with h5py.File(self.args.output, 'w') as outf:
+            # Merge the averages and stds
+            self._merge_data(outf)
+
+            # Merge runtime stats
+            self._merge_runtime_stats()
+
+            # Merge the metadata
+            self._merge_meta(outf)
+
+    @staticmethod
+    def _copy_pop(x, n=0):
+        xcopy = x.copy()
+        return xcopy.pop(n), xcopy
+
+    def _eval_fun_exprs(self, avars):
+        # Prepare the substitution dictionary
+        subs = dict(zip(self.anames, avars))
+
+        # Evaluate the function and return
+        return np.array([npeval(v, subs) for v in self.fexprs])
+
+    def _init_tavg_merge(self):
+        (file0, stats0, cfg0, dt0), f = self._copy_pop(self.files)
+        self.stats, self.cfg = stats0, cfg0
+
+        self.fields = fields = stats0.get('data', 'fields').split(',')
+        self.cfgsect = cfgsect = stats0.get('tavg', 'cfg-section')
+        self.region = cfg0.get(cfgsect, 'region')
+        self.uuid = file0['mesh_uuid']
+
+        self.dkeys = [k for k in file0 if k.startswith('tavg_')]
+
+        # Mapping for expression to sorted list
+        aidx = [f.startswith('avg-') for f in fields]
+        sidx = [f.startswith('std-') for f in fields]
+        self.anames = [f.removeprefix('avg-') for f in compress(fields, aidx)]
+        self.snames = [f.removeprefix('std-') for f in compress(fields, sidx)]
+        self.mapping = [(aidx, np.argsort(self.anames),
```

Comment:
Asking NumPy to sort lists is usually asking for trouble. `sorted(range(...), key=lambda i: ...)` is what you want.